### PR TITLE
`Plagiarism cases`: Send emails for notifications posts in plagiarism cases

### PR DESCRIPTION
**Pull request by JohannesStoehr**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Server
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/server/).
- [x] I added multiple integration tests (Spring) related to the features (with a high test coverage).
- [x] I implemented the changes with a good performance and prevented too many database calls.
- [x] I documented the Java code using JavaDoc style.

### Motivation and Context

Currently students are not notified via email that they received a plagiarism case, even though they should. 

### Description

This PR adds email templates and makes sure that they’re sent when a plagiarism case is notified. 

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Student
- Your own account
- 1 Exercise

1. Log in to Artemis
2. Participate with both students (your account and other student)
3. Check for plagiarism in that exercise (you might have to decrease the needed similarity)
4. Confirm the case for your user
5. Notify your user account about the case in the plagiarism overview of the course
6. Set a verdict
7. Check the emails in your mail client

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage

| Class/File | Branch | Line |
|------------|-------:|-----:|
| SingleUserNotificationService.java | 93% | 97% |

### CODE REVIEW

**Review by yuling-sun — DISMISSED** [review]
Tested on TS1. Works as described.

**Review by SimonPaul02 — APPROVED** [review]
Tested on TS1, works as described

**Comment by akesfeden** [comment]
I don't know much about the email related parts of the project, are emails always in English? I couldn't see German translations for the strings in [messages.properties](https://github.com/ls1intum/Artemis/pull/5565/files#diff-e9d91b0ce71b56fa7bbe03d9addcd6434088a3a1ee27d0b8feda0de045282101) or [PlagiarismVerdict.java](https://github.com/ls1intum/Artemis/pull/5565/files#diff-50506eccb73cd0e4ce09c568303b6f2877e4c206adcee8c227eb7c9f19364cd7) that's why I asked.

**Comment by JohannesStoehr** [comment]
> I don't know much about the email related parts of the project, are emails always in English? I couldn't see German translations for the strings in [messages.properties](https://github.com/ls1intum/Artemis/pull/5565/files#diff-e9d91b0ce71b56fa7bbe03d9addcd6434088a3a1ee27d0b8feda0de045282101) or [PlagiarismVerdict.java](https://github.com/ls1intum/Artemis/pull/5565/files#diff-50506eccb73cd0e4ce09c568303b6f2877e4c206adcee8c227eb7c9f19364cd7) that's why I asked.

You're right I missed that. Now the student can also receive their plagiarism case in German

**Review by luisa-we — DISMISSED** [review]
Tested on TS1, works as expected.

**Review by Mikael31 — DISMISSED** [review]
Tested on ts1. everything worked as expected

**Review by Santia-go — DISMISSED** [review]
Tested on TS1 
Working as expected. The email is received immediately and it is possible to access to the case through the "Open in Artemis" button. 
Screenshot 2022-08-15 at 16 14 46

File: src/main/resources/i18n/messages.properties [hunk 1]
```diff
@@ -90,3 +92,11 @@ email.notification.aux.exercise.type.modeling=The modeling exercise
 email.notification.aux.difficulty.easy=Easy
 email.notification.aux.difficulty.medium=Medium
 email.notification.aux.difficulty.hard=Hard
+
+# Plagiarism
+email.notification.title.post.plagiarismVerdict=New plagiarism verdict in exercise {0}
```
Comment (ge65cer) [inline src/main/resources/i18n/messages.properties:97]:
```suggestion
email.notification.title.post.plagiarismVerdict=Verdict reached in plagiarism case for exercise {0}
```

File: src/main/resources/i18n/messages_de.properties [hunk 2]
```diff
@@ -92,3 +92,11 @@ email.notification.aux.exercise.type.modeling=Die Modellieraufgabe
 email.notification.aux.difficulty.easy=Leicht
 email.notification.aux.difficulty.medium=Mittel
 email.notification.aux.difficulty.hard=Schwer
+
+# Plagiarism
+email.notification.title.post.plagiarismVerdict=Neue Entscheidung zu dem Plagiat in Aufgabe {0}
```
Comment (ge65cer) [inline src/main/resources/i18n/messages_de.properties:97]:
```suggestion
email.notification.title.post.plagiarismVerdict=Entscheidung zum Plagiatsfall in der Aufgabe {0} gefallen
```

File: src/main/resources/i18n/messages_en.properties [hunk 3]
```diff
@@ -92,3 +92,11 @@ email.notification.aux.exercise.type.modeling=The modeling exercise
 email.notification.aux.difficulty.easy=Easy
 email.notification.aux.difficulty.medium=Medium
 email.notification.aux.difficulty.hard=Hard
+
+# Plagiarism
+email.notification.title.post.plagiarismVerdict=New plagiarism verdict in exercise {0}
```
Comment (ge65cer) [inline src/main/resources/i18n/messages_en.properties:97]: As above

**Review by ge65cer — CHANGES_REQUESTED** [review]
Tested on TS1, e-mails are sent for both the notification and the verdict. 👍  I have the following feedback:
* Render the Markdown in the notification body correctly
* Include the verdict information (percentage of point deduction or actual warning message) in the email
* Some small text suggestions

**Comment by JohannesStoehr** [comment]
> Render the Markdown in the notification body correctly

I'll cover this in a follow up, where I refactor the emails. Otherwise we have to render the markdown in multiple places

> Include the verdict information (percentage of point deduction or actual warning message) in the email

The email is mostly a notification that there is a verdict. But we could make the email more informative in the future

**Review by ge65cer — DISMISSED** [review]
@JohannesStoehr In that case, I can approve this PR.

**Review by julian-christl — DISMISSED** [review]
The code changes look reasonable and consistent.

**Review by ge65cer — APPROVED** [review]
Re-approve

File: src/main/resources/i18n/messages_en.properties [hunk 4]
```diff
@@ -92,3 +92,11 @@ email.notification.aux.exercise.type.modeling=The modeling exercise
 email.notification.aux.difficulty.easy=Easy
 email.notification.aux.difficulty.medium=Medium
 email.notification.aux.difficulty.hard=Hard
+
+# Plagiarism
+email.notification.title.post.plagiarismVerdict=Verdict reached in plagiarism case for exercise {0}
+email.notification.title.post.plagiarismCase=New plagiarism case in exercise {0}
+email.notification.aux.plagiarismVerdict.plagiarism=Your case is considered as plagiarism!
+email.notification.aux.plagiarismVerdict.point.deduction=Due to your plagiarism, we are deducting points from your score in this exercise!
+email.notification.aux.plagiarismVerdict.warning=Due to the plagiarism attempt, you are warned. Any further plagiarism will lead to severe consequences!
```
Comment (jpbernius) [inline src/main/resources/i18n/messages_en.properties:101]: Translation is inconsistent with German message:
> Weitere Plagiate können schwere Folgen haben!

vs
> Any further plagiarism will lead to severe consequences!

**Review by jpbernius — CHANGES_REQUESTED** [review]

**Review by julian-christl — APPROVED** [review]
Reapprove code

**Review by ge65cer — APPROVED** [review]

**Review by jpbernius — APPROVED** [review]

**Pull request merged**